api-stat.go

```diff
@@ -78,27 +79,37 @@ func (c Client) statObject(ctx context.Context, bucketName, objectName string, o
 	if err := s3utils.CheckValidObjectName(objectName); err != nil {
 		return ObjectInfo{}, err
 	}
+	headers := opts.Header()
+	if opts.ReplicationDeleteMarker {
+		headers.Set(minIOBucketReplicationDeleteMarker, "true")
+	}
 
 	urlValues := make(url.Values)
 	if opts.VersionID != "" {
 		urlValues.Set("versionId", opts.VersionID)
 	}
-
 	// Execute HEAD on objectName.
 	resp, err := c.executeMethod(ctx, http.MethodHead, requestMetadata{
 		bucketName:       bucketName,
 		objectName:       objectName,
 		queryValues:      urlValues,
 		contentSHA256Hex: emptySHA256Hex,
-		customHeader:     opts.Header(),
+		customHeader:     headers,
 	})
 	defer closeResponse(resp)
 	if err != nil {
 		return ObjectInfo{}, err
 	}
+	deleteMarker := false
+	if strings.ToLower(resp.Header.Get(amzDeleteMarker)) == "true" {
+		deleteMarker = true
+	}
```

Comment:
One line

```
deleteMarker := resp.Header.Get(amzDeleteMarker) == "true"
```

I don't know why would you need strings.ToLower since the value is not set externally!